src/aiortc/contrib/media.py

```diff
@@ -641,36 +762,40 @@ async def __run_track(self, track, context):
+                        source_frame_array = frame.to_rgb().to_ndarray()
+                        
+                        time_before_keypoints = time.perf_counter()
+                        with concurrent.futures.ThreadPoolExecutor() as pool:
+                            source_keypoints, _  = await loop.run_in_executor(pool, 
+                                                model.extract_keypoints, source_frame_array)
+                        time_after_keypoints = time.perf_counter()
+                        self.__log_debug("Source keypoints extraction time in receiver: %s",
+                                        str(time_after_keypoints - time_before_keypoints))
+
+                        asyncio.run_coroutine_threadsafe(self.__source_queue.put(
+                            (source_frame_array, source_keypoints, video_frame_index)), loop)
                     
-                    time_before_keypoints = time.perf_counter()
-                    with concurrent.futures.ThreadPoolExecutor() as pool:
-                        source_keypoints, _  = await loop.run_in_executor(pool, 
-                                            model.extract_keypoints, source_frame_array)
-                    time_after_keypoints = time.perf_counter()
-                    self.__log_debug("Source keypoints extraction time in receiver: %s",
-                                    str(time_after_keypoints - time_before_keypoints))
-
-                    asyncio.run_coroutine_threadsafe(self.__video_queue.put((source_frame_array, source_keypoints, video_frame_index)), loop)
-
+                    elif self.__display_option == "real":
+                        # directly display the received real video frames
+                        asyncio.run_coroutine_threadsafe(self.__real_video_queue.put(
+                                                        (frame, video_frame_index))
+                                                        , loop)
                 else:
-                    # regular video stream
-                    self.__log_debug("Received original video frame %s with index %s at time %s",
-                                    frame, video_frame_index, datetime.datetime.now())
-                    self.__setsize(track)
-                        
+                    # Original aiortc video stream, no prediction
                     if self.__recv_times_file is not None:
                         self.__recv_times_file.write(f'Received {video_frame_index} at {datetime.datetime.now()}\n')
                         self.__recv_times_file.flush()
 
                     if self.__save_dir is not None:
-                        frame_array = frame.to_rgb().to_ndarray()
-                        np.save(os.path.join(self.__save_dir, 
-                            'receiver_frame_%05d.npy' % video_frame_index), frame_array)
+                        pass
+                        #frame_array = frame.to_rgb().to_ndarray()
```

Comment:
why this removal of saving the original frame? it is necessary for experiments

Reply:
Saving was taking so much time and space for the task I needed. I reverted back the savings